plugins/outputs/warp10/warp10.go

```diff
@@ -0,0 +1,293 @@
+				log.Printf("E! [outputs.warp10] Could not encode value: %v", err)
+				continue
+			}
+			metric.Value = metricValue
+
+			tagsSlice := buildTags(mm.TagList())
+			metric.Tags = strings.Join(tagsSlice, ",")
+
+			messageLine := fmt.Sprintf("%d// %s{%s} %s\n", metric.Timestamp, metric.Metric, metric.Tags, metric.Value)
+
+			collectString = append(collectString, messageLine)
+		}
+	}
+	return fmt.Sprint(strings.Join(collectString, ""))
+}
+
+// Write metrics to Warp10
+func (w *Warp10) Write(metrics []telegraf.Metric) error {
+
+	if len(metrics) == 0 {
+		return nil
+	}
+	var now = time.Now()
+	payload := w.GenWarp10Payload(metrics, now)
+
+	req, err := http.NewRequest("POST", w.WarpURL+"/api/v0/update", bytes.NewBufferString(payload))
+	req.Header.Set("X-Warp10-Token", w.Token)
+	req.Header.Set("Content-Type", "text/plain")
+
+	resp, err := w.client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
+		return fmt.Errorf(w.WarpURL + ": " + w.HandleError(string(body)))
+	}
+
+	return nil
+}
+
+func buildTags(tags []*telegraf.Tag) []string {
+
+	tagsString := make([]string, len(tags)+1)
+	indexSource := 0
+	for index, tag := range tags {
+		tagsString[index] = fmt.Sprintf("%s=%s", tag.Key, tag.Value)
+		indexSource = index
+	}
+	indexSource++
+	tagsString[indexSource] = fmt.Sprintf("source=telegraf")
+	sort.Strings(tagsString)
+	return tagsString
+}
+
+func buildValue(v interface{}) (string, error) {
+	var retv string
+	switch p := v.(type) {
+	case int64:
+		retv = intToString(int64(p))
+	case string:
+		retv = fmt.Sprintf("'%s'", strings.Replace(p, "'", "\\'", -1))
+	case bool:
+		retv = boolToString(bool(p))
+	case uint64:
+		retv = uIntToString(uint64(p))
+	case float64:
+		retv = floatToString(float64(p))
+	default:
+		retv = "'" + strings.Replace(fmt.Sprintf("%s", p), "'", "\\'", -1) + "'"
+	}
+	return retv, nil
+}
+
+func intToString(inputNum int64) string {
+	return strconv.FormatInt(inputNum, 10)
+}
+
+func boolToString(inputBool bool) string {
+	return strconv.FormatBool(inputBool)
+}
+
+func uIntToString(inputNum uint64) string {
+	return strconv.FormatUint(inputNum, 10)
+}
+
+func floatToString(inputNum float64) string {
+	return strconv.FormatFloat(inputNum, 'f', 6, 64)
+}
+
+// SampleConfig get config
+func (w *Warp10) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description get description
+func (w *Warp10) Description() string {
+	return "Configuration for Warp server to send metrics to"
+}
+
+// Close close
+func (w *Warp10) Close() error {
+	return nil
+}
+
+func init() {
+	outputs.Add("warp10", func() telegraf.Output {
+		return &Warp10{}
+	})
+}
+
+// HandleError read http error body and return a corresponding error
+func (w *Warp10) HandleError(body string) string {
+	if body == "" {
+		return "Empty return"
+	}
+
+	if strings.Contains(body, "Invalid token") {
+		return fmt.Sprintf("Invalid token: %v", w.Token)
```

Comment:
Don't include the token in any errors because then it will be logged, which we don't want to do with any secrets.